src/include/buffer.h

```diff
@@ -79,6 +79,23 @@ template<typename T> class DencDumper;
 
 namespace ceph {
 
+#ifdef BUFFER_CORRUPTION_DEBUG
+class canary_t {
+  std::uint64_t magic{0xBADC0FFEEBADC0DE};
+public:
+  void is_alive_or_die() const {
+    if (magic != 0xBADC0FFEEBADC0DE) {
```

Comment:
how about unlikely ?

Reply:
Hmm, not sure we need to take care about performance in debug arrangements. Is there any reason apart that?

Reply:
I was mostly thinking about as little disturbance to existing timings as possible. This assumes that you do have a suspicion on some overwrites and you enable canaries. It would be shame if overwrites went away.